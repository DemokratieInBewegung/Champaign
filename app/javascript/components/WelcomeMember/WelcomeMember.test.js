// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WelcomeMember from './WelcomeMember';

const fn = jest.fn();

const member = {
  id: 1,
  name: 'Jane Roe',
  email: 'janeroe@example.com',
  registered: false,
  donorStatus: 'donor',
};

test('renders correctly when given a valid member', () => {
  const wrapper = shallow(<WelcomeMember member={member} resetMember={fn} />);
  expect(toJson(wrapper)).toMatchSnapshot('WelcomeMemberWithMember');
});

test('does not render if the member is null', () => {
  const wrapper = shallow(<WelcomeMember member={null} resetMember={fn} />);
  expect(toJson(wrapper)).toMatchSnapshot('WelcomeMemberWithoutMember');
  expect(wrapper.get(0)).toBeNull();
});
