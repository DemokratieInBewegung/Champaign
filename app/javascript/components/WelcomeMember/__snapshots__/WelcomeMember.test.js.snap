// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeMemberWithMember 1`] = `
<div
  className="WelcomeMember"
>
  <i
    className="WelcomeMember__icon fa fa-check-square-o"
  />
  <div>
    <span
      className="WelcomeMember__name"
    >
      Jane Roe
    </span>
    <a
      className="WelcomeMember__link"
      onClick={[Function]}
    >
      <FormattedMessage
        id="form.switch_user"
        values={Object {}}
      />
    </a>
  </div>
</div>
`;

exports[`WelcomeMemberWithoutMember 1`] = `null`;
